### PR TITLE
Fix slowness when checking out spatial-filter WC

### PR DESCRIPTION
**Pull request by olsen232**
Slowness is due to libgit2 repeatedly refreshing whenever it encounters
a missing object, in the hopes that the missing object is only due to
its stale view of the data on disk. The fix is to refresh once before
the WC checkout and then disable refreshing for the rest of the
checkout - this current-as-of-start-of-checkout view of the ODB is
good enough for us since we don't read and write concurrently.

https://github.com/koordinates/kart/issues/561

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,10 @@ Please note that compatibility for 0.x releases (software or repositories) isn't
 
 _When adding new entries to the changelog, please include issue/PR numbers wherever possible._
 
+## 0.11.1
+
+* Fix for the slowness when creating the working copy in a spatially filtered repository (this was previously slower than usual in a spatially filtered repository, now it is faster) [#561](https://github.com/koordinates/kart/issues/561)
```
Comment (craigds) [inline CHANGELOG.md:9]:
```suggestion
* Improve performance when creating a working copy in a spatially filtered repository (this was previously slower than in a non-filtered repository; now it is much faster) [#561](https://github.com/koordinates/kart/issues/561)
```

File: kart/working_copy/base.py [hunk 2]
```diff
@@ -1444,3 +1445,10 @@ def _reset_dirty_rows(
             self._reset_tracking_table_for_table(
                 sess, base_ds.table_name, feature_filter
             )
+
+
+@contextlib.contextmanager
+def pause_refreshing(odb):
+    odb.set_lookup_flags(pygit2.GIT_ODB_LOOKUP_NO_REFRESH)
+    yield
+    odb.set_lookup_flags(0)
```
Comment (craigds) [inline kart/working_copy/base.py:1454]:
```suggestion
    old_flags = odb.lookup_flags()
    odb.set_lookup_flags(pygit2.GIT_ODB_LOOKUP_NO_REFRESH)
    try:
        yield
    finally:
        odb.set_lookup_flags(old_flags)
```

**Review by craigds — APPROVED** [review]
nice find

File: CHANGELOG.md (hunk 1 above)
Comment (olsen232) [reply CHANGELOG.md]: Done

File: kart/working_copy/base.py (hunk 2 above)
Comment (olsen232) [reply kart/working_copy/base.py]: Done

**Pull request merged**